docker/README.md

```diff
@@ -0,0 +1,91 @@
+# Using Docker for DEV
+
+Grid has a micro-service architecture which is great for separating concerns,
+however it does mean setting up a development environment can be a little confusing.
+
+[Docker](https://www.docker.com/) is well suited to this problem, allowing you to compose the
+structure of an app in a [yaml file](./docker-compose.yml).
+
+NB: this has **not** been tested in PROD.
+
+## Architecture
+There is a container for each service of the Grid.
+
+There is also a:
+- `configs` container which will generate all the required configs
+- `data` container which is a shared [volume container](https://docs.docker.com/engine/userguide/dockervolumes/#creating-and-mounting-a-data-volume-container).
+  This container shares the code across the containers and the `.ivy2` and `.sbt` directories
+  allowing the cached files to be shared between containers.
+- `nginx` container, a proxy server to each of the services
+
+## Requirements
+- Docker
+- Docker Compose
+
+### Additional requirements for OSX
+- [docker-osx-dev](https://github.com/brikis98/docker-osx-dev)
+
+## Setup
+- Ensure you've set up the aws-cli with a `media-service` profile (`aws configure --profile media-service`).
+- Create `$HOME/.gu/grid/grid-settings.yml` using [`grid-settings.yml.template`](./configs/generators/grid-settings.yml.template) as a template.
```

Comment:
By default, the docker-machine VM has [limited access](https://docs.docker.com/engine/userguide/dockervolumes/#mount-a-host-directory-as-a-data-volume) to the host file system, so the location changed from `/etc/hosts` to `$HOME/.gu`.

> If you are using Docker Machine on Mac or Windows, your Docker daemon has only limited access to your OS X or Windows filesystem. Docker Machine tries to auto-share your /Users (OS X) or C:\Users (Windows) directory.

> All other paths come from your virtual machine’s filesystem. For example, if you are using VirtualBox some other folder available for sharing, you need to do additional work. In the case of VirtualBox you need to make the host folder available as a shared folder in VirtualBox. Then, you can mount it using the Docker -v flag.

Alternatively, we could share `/etc/gu` from an OSX host to the Docker VM, however this seemed better as it means fewer setup steps.

Alternatively alternatively, we could use linux by default...